homeassistant/components/command_line/binary_sensor.py

```diff
@@ -133,17 +136,21 @@ async def _async_update(self) -> None:
         await self.data.async_update()
         value = self.data.value
 
+        variables = self._template_variables_with_value(value)
+        if not self._render_availability_template(variables):
+            return
+
         if self._value_template is not None:
-            value = self._value_template.async_render_with_possible_json_value(
-                value, None
+            value = self._value_template.async_render_as_value_template(
+                self.entity_id, variables, None
             )
         self._attr_is_on = None
         if value == self._payload_on:
             self._attr_is_on = True
         elif value == self._payload_off:
             self._attr_is_on = False
 
-        self._process_manual_data(value)
+        self._process_manual_data(variables)
```

Comment:
Why do we now pass the `variables` instead of the rendered `value`? Same question for the other changed modules.

Reply:
Because, before we were creating 2 sets of the same exact variable data.  This means we took the value and converted it to json twice every update for all entities that used value_template.